lib/Driver/Job.cpp

```diff
@@ -240,6 +240,10 @@ StringRef CommandOutput::getAnyOutputForType(types::ID Type) const {
   return getAdditionalOutputForType(Type);
 }
 
+const OutputFileMap &CommandOutput::getDerivedOutputMap() const {
+  return DerivedOutputMap;
```

Comment:
Hm, is the Job really the right way to expose this? It's a shared structure and will contain more information than just what's in the Job.

Reply:
I don't really like that this whole function is file-static. Maybe if you can help me figure out where it really belongs this issue will be clearer. I suspect I did it that way because it represented the least amount of encapsulation-breaking needed, given where the client was.

Reply:
What did we decide to do about this?

Reply:
Maybe the whole "write the subset of the DOM based on this Job's inputs" should be pushed into Job (taking an `llvm::raw_ostream &`). What do you think?